File: 500.py
class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        rows = [
            set("qwertyuiop"),
            set("asdfghjkl"),
            set("zxcvbnm")
        ]
        mapper = {}
        for set_ in rows:
            for k in set_:
                mapper[k] = set_
                
        def ok(w):
            set_ = mapper[w[0].lower()]
            return all(k in set_ for k in w.lower())

        return list(filter(ok, words))

File: src/500.rs
use std::collections::{HashMap, HashSet};

struct Solution {}

impl Solution {
    pub fn find_words(words: Vec<String>) -> Vec<String> {
        let rows : Vec<HashSet<char>> = vec![
            "qwertyuiop".to_string().chars().collect(),
            "asdfghjkl".to_string().chars().collect(),
            "zxcvbnm".to_string().chars().collect(),
        ];
        let mut mapper = HashMap::new();
        for row in rows.iter() {
            for ch in row.iter() {
                mapper.insert(ch, row);
            }
        }
        words.into_iter()
            .filter(|word| {
                let first_char = word.chars().next().unwrap();
                let first_char_lower_case = first_char.to_lowercase().collect::<Vec<char>>()[0];
                let row_set = &mapper[&first_char_lower_case];
                word.to_lowercase().chars().all(|ch| {
                    row_set.contains(&ch)
                })
            })
            .collect()
    }
}

